File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: bool
        """
        slow = head
        fast = head 
        while fast and fast.next :
            slow = slow.next 
            fast = fast.next.next 
        middle = slow
        slow = None
        cur = middle
        fast = cur 
        while fast :
            fast = cur.next 
            cur.next = slow 
            slow = cur
            cur = fast 
        cur = head

        while slow :
            if  slow.val != cur.val :
                return False
            slow = slow.next
            cur = cur.next 
        return True
        
        
        

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        ArrayList<Integer> l = new ArrayList<>();
        ListNode temp = head ;
        while(temp != null){
            l.add(temp.val);
            temp = temp.next ;
        }
        int n = l.size();
        for(int j = 0 ; j < n / 2 ; j++ ){
            if(!(l.get(j) == l.get(n-j-1))) return false;
        }
        return true;
        

        
    }
}